{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/training/notebooks/Train.py

```diff
@@ -1,140 +1,137 @@
 
-r = Recipe(profile=profile)
+input_table_path = dbutils.widgets.get("training_data_path")
+experiment_name = dbutils.widgets.get("experiment_name")
+model_name = dbutils.widgets.get("model_name")
 
 # COMMAND ----------
+# DBTITLE 1, Set experiment
 
-r.clean()
-
-# COMMAND ----------
+import mlflow
 
-r.inspect()
+mlflow.set_experiment(experiment_name)
 
 # COMMAND ----------
+# DBTITLE 1, Load raw data
 
-r.run("ingest")
+training_df = spark.read.format("delta").load(input_table_path)
+display(training_df)
 
 # COMMAND ----------
+# DBTITLE 1, Helper function
+from mlflow.tracking import MlflowClient
+import mlflow.pyfunc
 
-r.run("split")
 
-# COMMAND ----------
+def get_latest_model_version(model_name):
+    latest_version = 1
+    mlflow_client = MlflowClient()
+    for mv in mlflow_client.search_model_versions(f"name='{model_name}'"):
+        version_int = int(mv.version)
+        if version_int > latest_version:
+            latest_version = version_int
+    return latest_version
 
-r.run("transform")
 
 # COMMAND ----------
 
-r.run("train")
+# MAGIC %md
+# MAGIC Train a LightGBM model on the data, then log and register the model with MLflow.
 
 # COMMAND ----------
+# DBTITLE 1, Train model
 
-r.run("evaluate")
+import mlflow
+from sklearn.model_selection import train_test_split
+import lightgbm as lgb
+import mlflow.lightgbm
 
-# COMMAND ----------
+# Collect data into a Pandas array for training. Since the timestamp columns would likely
+# cause the model to overfit the data, exclude them to avoid training on them.
+columns = [col for col in training_df.columns if col not in ['tpep_pickup_datetime', 'tpep_dropoff_datetime']]
+data = training_df.toPandas()[columns]
 
-r.run("register")
+train, test = train_test_split(data, random_state=123)
+X_train = train.drop(["fare_amount"], axis=1)
+X_test = test.drop(["fare_amount"], axis=1)
+y_train = train.fare_amount
+y_test = test.fare_amount
 
-# COMMAND ----------
+mlflow.lightgbm.autolog()
+train_lgb_dataset = lgb.Dataset(X_train, label=y_train.values)
+test_lgb_dataset = lgb.Dataset(X_test, label=y_test.values)
 
-r.inspect("train")
+param = {"num_leaves": 32, "objective": "regression", "metric": "rmse"}
+num_rounds = 100
+
+# Train a lightGBM model
+model = lgb.train(param, train_lgb_dataset, num_rounds)
 
 # COMMAND ----------
+# DBTITLE 1, Log model and return output.
 
-test_data = r.get_artifact("test_data")
-test_data.describe()
+# Log the trained model with MLflow
+mlflow.lightgbm.log_model(
+    model, artifact_path="lgb_model", registered_model_name=model_name,
+)
 
-# COMMAND ----------
+# Build out the MLflow model registry URL for this model version.
+workspace_url = spark.conf.get("spark.databricks.workspaceUrl")
+model_version = get_latest_model_version(model_name)
+model_registry_url = "https://{workspace_url}/#mlflow/models/{model_name}/versions/{model_version}".format(
+    workspace_url=workspace_url, model_name=model_name, model_version=model_version
+)
```

Comment:
where is this used?

Reply:
I don't believe it is, but it's the same in the `TrainWithFeatureStore.py` notebook that this is modeled on. I believe we create this URL here so the user is aware of where the version is/what the URL looks like.

Reply:
Ah, we may have returned it as a GH PR comment in the past.  If it's not used though, we could clean it up.

Reply:
Cleaned it up!